src/automations/Job.ts

```diff
@@ -33,22 +35,25 @@ export default class Job {
     public async clonePost(
         posts: PostInfo[],
         regionsToPost: string[],
-        sourceID: number
+        sourceID: number,
     ) {
-        const spinner = ora().start(`Starting to create job posts.`);
+        this.spinner.start(`Starting to create job posts.`);
+        const board =  PROTECTED_JOB_BOARDS[0];
         let protectedPosts: PostInfo[];
         // Check if a source post is provided.
         if (sourceID) {
-            protectedPosts = posts.filter((post) => post.id === sourceID);
+            protectedPosts = posts.filter((post) => post.id === sourceID && post.boardInfo.name === board);
         } else {
             // If a source post is not provided, use posts that are in the "Canonical" board.
             protectedPosts = posts.filter(
-                (post) => post.boardInfo.name === PROTECTED_JOB_BOARDS[0]
+                (post) => post.boardInfo.name === board
             );
         }
 
-        if (!protectedPosts || !protectedPosts.length)
-            throw new Error(`No post found to clone`);
+        if (!protectedPosts || !protectedPosts.length) {
```

Comment:
You can do something cool using the optional chaining here:
```js
if (!protectedPosts?.length)
```

Reply:
Done :+1: